### ISSUE TITLE
Windows: NCPA 1.7.2 - In some how it stops to function

### ISSUE THREAD

**Issue by varnar** [NagiosEnterprises/ncpa#86]
Hello 
I want let you know that in some how NCPA Listener has been stopped to function.
The process is show as running but it does nothing.
```
C:\Windows\system32>tasklist | find "ncpa"
ncpa_listener.exe             1476 Services                   0     52,208 K
```
In the log file is already 2 days no any records:
```
2014-10-18 00:44:22,118:DEBUG:handler:Initializing WebSocket
2014-10-18 00:44:22,118:DEBUG:handler:Validating WebSocket request
2014-10-18 00:44:22,118:DEBUG:__init__:Getting accessor: agent/plugin/Check_IISv7_AppPool_State.vbs/survey.questrade.com
2014-10-18 00:44:22,118:DEBUG:__init__:Running process with command line: `cscript C:\Program Files (x86)\Nagios\NCPA\plugins\Check_IISv7_AppPool_State.vbs survey.xxxx.com //NoLogo`
2014-10-18 00:44:22,259:INFO:handler:10.98.6.44 - - [2014-10-18 00:44:22] "GET /api/agent/plugin/Check_IISv7_AppPool_State.vbs/survey.xxxx.com?token=xxxxxx&check=1 HTTP/1.0" 200 306 0.140000
```

To have it back to working, you need to kill process, because service became not accessible.
![image](https://cloud.githubusercontent.com/assets/7560646/4767650/53d9eece-5b60-11e4-8439-39a320cb7704.png)

```
for /F "tokens=1,2" %i in ('"tasklist | FIND "ncpa_listener.exe""') do ( (TASKKILL /F /PID %j) ) & net start "NCPA Listener ncpalistener"
```

I have created a dump file, but since GIT does not allow attache binary file, I have RAR it and changed extension to JPG, so please change to RAR back and un-rar it.
If you need additional information - please let me know, I will try to collect as much as I can.

https://cloud.githubusercontent.com/assets/7560646/4767592/bc58a59a-5b5f-11e4-9cb1-0b80de45ecee.jpg rename to ncpa_listener.part1.rar

https://cloud.githubusercontent.com/assets/7560646/4767594/bf1e42da-5b5f-11e4-8d4e-da798cd7aa9f.jpg rename to ncpa_listener.part2.rar

![ncpa_listener part1](https://cloud.githubusercontent.com/assets/7560646/4767592/bc58a59a-5b5f-11e4-9cb1-0b80de45ecee.jpg)

![ncpa_listener part2](https://cloud.githubusercontent.com/assets/7560646/4767594/bf1e42da-5b5f-11e4-8d4e-da798cd7aa9f.jpg)

**Comment by varnar** [comment]
@scot0357, I have question.
Did you found why it don't return?
How it can be that the same script that executed using NSClient++ has no any issue such that?
Can you please some how check why it can stuck?
You have resolved issue when some executions are going to stuck and you avoid the server from halting.
But the root cause - why it stuck? And looks like we did not found it.

Thanks
Happy New Year

**Comment by scot0357** [comment]
I'm thinking its a permissions issue, NSClient might be running with adifferent permissions than NCPA, I'm formulating a test in my VM environment to see if I duplicate it, haven't had any luck so far though.

**Comment by SuperGoodMike** [comment]
@ Varnar - Does the script(s) run fine for a week or two then one day it will get stuck all of a sudden?

**Comment by varnar** [comment]
Yes, @SuperGoodMike, the scripts does run fine for couple of weeks and one day they start to stuck.

**Comment by scot0357** [comment]
Varnar,

After tracking this down, I'm quite sure this is caused by processes getting run by NCPA and them simply never returning and hitting some hidden timeout limit I've yet to find and it not being handled properly. Why they are causing issue, I'm still unsure, however I believe the concurrency fix noted above will help fix this issue as that 100% explains why the server hangs indefinitely. I'm not going to close this, but I cannot reproduce this issue in order to effectively troubleshoot it.

However, at this point I am a bit confused, does it _sometimes_ execute the plugin successfully? Or does it always fail, but sometimes it hangs?

**Comment by varnar** [comment]
At the couple of weeks after NCPA has been installed on that server the plugin executed successfully.
I can provide the initial plugin as well, if you need it.
Thanks

**Comment by scot0357** [comment]
Hmm, so I gather from that it has failed to execute a few times, but otherwise executes properly. When it does fail, it fails really hard to takes the whole agent down with it. This is confirming my beliefs that the subprocess fix above should remedy this issue. Is that `//T:5` forcing a timeout? Can you confirm that it is indeed enforcing that timeout? I'm quite sure that the subprocess fix getting rolled out in 1.8.0 will fix this, I'm interested as to why the subprocess was not dieing.

**Comment by varnar** [comment]
Right now I cannot try it, because the issue is happening on one of our Production server and I don't want to stopping NCPA agent doing that test :)

**Comment by jhomann** [comment]
Any update on this bug? Would like to know if it's been fixed or not.

**Comment by jhomann** [comment]
I'm going to close this, feel free to open it again if there is anything more you can tell us as @scot0357 has mentioned that this should be fixed in 1.8.0+.
